packages/core/mesh/teleport/src/rpc-extension.ts

```diff
@@ -17,15 +17,15 @@ export abstract class RpcExtension<Client, Server> implements TeleportExtension
   constructor(private readonly _rpcParams: Omit<ProtoRpcPeerOptions<Client, Server>, 'port' | 'handlers'>) {}
 
   get initiator() {
-    return this._extensionContext.initiator;
+    return this._extensionContext?.initiator;
```

Comment:
not feeling confident about this change, but made it just to reduce noise in logs. fixes property access on undefined for when a teleport connection was aborted during opening.
connection registration cannot be easily synchronized with teleport destroy and abort because it's not done inside teleport instance
```ts
export const createTeleportProtocolFactory = (
  onConnection: (teleport: Teleport) => Promise<void>,
  defaultParams?: Partial<TeleportParams>,
): WireProtocolProvider => {
  return (params) => {
    const teleport = new Teleport({ ...defaultParams, ...params });
    return {
      stream: teleport.stream,
      open: async (sessionId?: PublicKey) => {
        await teleport.open(sessionId);
        await onConnection(teleport); // here
      }
      ...
};
```